frontend/src/main/web/src/components/EditableText.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component, PropTypes } from 'react'
+import { merge } from 'lodash'
+import {
+  Row,
+  TextInput
+} from './'
+
+const classes = {
+  root: {
+    w: 'W(100%)'
+  },
+  textInput: {
+    base: {
+      ai: 'Ai(c)',
+      bgc: 'Bgc(#fff)',
+      h: 'H(r1h)',
+      w: 'W(100%)'
+    }
+  },
+  text: {
+    base: {
+      ai: 'Ai(c)',
+      bd: 'Bd(bd2) Bdc(t)',
+      c: 'Cur(t)',
+      h: 'H(r1h)',
+      lineClamp: 'LineClamp(1,36px)',
+      p: 'Px(rq) Py(re)',
+      w: 'W(100%)'
+    },
+    editable: {
+      brds: 'Bdrs(rq)',
+      trs: 'Trs(aeo)',
+      hover: {
+        bd: 'editable:h_Bd(bd2) editable:h_Bdc(neutral)'
+      }
+    },
+    placeholder: {
+      c: 'C(muted)'
+    }
+  }
+}
+
+class EditableText extends Component {
+  constructor () {
+    super()
+    this.state = {
+      focus: false
+    }
+  }
+  handleClick () {
+    this.setState({ focus: true })
```

Comment:
This component is text input that switches to label when its not focused. 
This line will switch the from label to text field and set it on focus.